### PR TITLE
Use default max results of 10000 for model registry search_model_versions() API

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Use default max results of 10000 for model registry search_model_versions() API for compatibility with other backends, e.g Databricks

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

## Release Notes

Fix a bug causing too many results to be requested by default when calling ``MlflowClient.search_model_versions()``

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/model_registry/__init__.py [hunk 1]
```diff
@@ -3,5 +3,5 @@
 
 SEARCH_REGISTERED_MODEL_MAX_RESULTS_DEFAULT = 100
 SEARCH_REGISTERED_MODEL_MAX_RESULTS_THRESHOLD = 1000
-SEARCH_MODEL_VERSION_MAX_RESULTS_DEFAULT = 200_000
+SEARCH_MODEL_VERSION_MAX_RESULTS_DEFAULT = 1000
```
[the remark this answers is not in the document]
Comment (dbczumar) [reply mlflow/store/model_registry/__init__.py:9]: Yes, SearchRuns only supports 10,000

File: mlflow/store/model_registry/__init__.py [hunk 2]
```diff
@@ -3,5 +3,5 @@
...
 SEARCH_MODEL_VERSION_MAX_RESULTS_THRESHOLD = 200_000
```
[the remark this answers is not in the document]
Comment (dbczumar) [reply mlflow/store/model_registry/__init__.py:10]: No, other backends support 200k and always have

**Review by BenWilson2 — APPROVED** [review]
LGTM

**Review by jerrylian-db — APPROVED** [review]
LGTM

**Comment by tkb-cdl** [comment]
Our application code pins version 2.2.0 of mlflow. Now, calling `search_model_versions` with the default arguments is rejected by the server, with message: `RestException: INVALID_PARAMETER_VALUE: Invalid max results 200000, should be between 0 and 10000`. Presumably this is because the databricks-hosted tracking servers were updated to 2.2.1.

Did we miss a deprecation warning? I didn't expect this function to break in a point update.

**Pull request merged**